### PR TITLE
[#51] Updated federation documentation (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: docs/system_overview/federation.md [hunk 1]
```diff
@@ -9,22 +9,76 @@ Primary reasons for using Zone Federation include:
...
-ZoneA $ iadmin mkzone ZoneB remote zoneB-iCAT.hostname.example.org:ZoneBPort
+ZoneA $ iadmin mkzone ZoneB remote zoneB-provider.example.org:2247
 ZoneA $ iadmin mkuser bobby#ZoneB rodsuser
 ~~~
 
 In Zone B, add Zone A, but skip adding any remote users at this time:
 
 ~~~
-ZoneB $ iadmin mkzone ZoneA remote zoneA-iCAT.hostname.example.org:ZoneAPort
+ZoneB $ iadmin mkzone ZoneA remote zoneA-provider.example.org:1247
```
Comment (trel) [inline docs/system_overview/federation.md:81]: same...
`ZoneB $ iadmin mkzone ZoneA remote`

File: docs/system_overview/federation.md [hunk 2]
```diff
@@ -9,22 +9,76 @@ Primary reasons for using Zone Federation include:
...
 ~~~
 
 Then, any user of Zone A will be able to grant permissions to `bobby#ZoneB`:
```
Comment (trel) [inline docs/system_overview/federation.md:84]: `ZoneA`

**Comment by trel** [comment]
i also see reference to 'resource server' on that page, can you take a pass and make sure everything talks about 'provider' and 'cosumer' server?

File: docs/system_overview/federation.md [hunk 3]
```diff
@@ -9,22 +9,76 @@ Primary reasons for using Zone Federation include:
...
+        "zone_key": "ZONE_KEY_FOR_ZONE_A", 
+        "zone_name": "ZoneA", 
+        "zone_port": 1247
+    }
+]
+~~~
+
+For more information, see [Server Authentication](#server-authentication).
+
+It is important that the servers are able to resolve the host name of the federated zone. This can be handled in many ways. The simpliest solution is to add an entry to the `/etc/hosts` file.
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply docs/system_overview/federation.md:67]: We can remove it or change it if you like. I wanted to add a line that reminds users to make sure that the hostnames can be resolved. This is really easy to run into when setting up docker containers. I encountered this many times :)

[the remark this answers is not in the document]
Comment (trel) [reply docs/system_overview/federation.md:67]: definitely understood.   try swapping the last sentence with mine above and we'll read it again for the first time after you push.

File: docs/system_overview/federation.md [hunk 4]
```diff
@@ -9,22 +9,76 @@ Primary reasons for using Zone Federation include:
...
+
+With the server configuration and networking complete, all that is left is to define the remote zone and users in the catalog.
 
 In Zone A, add Zone B and define a remote user:
 
 ~~~
-ZoneA $ iadmin mkzone ZoneB remote zoneB-iCAT.hostname.example.org:ZoneBPort
+ZoneA $ iadmin mkzone ZoneB remote zoneB-provider.example.org:2247
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply docs/system_overview/federation.md:74]: The connection info is required when creating the zone. I get the following on the command line:
```bash
remote addresses: XXX.XXX.XXX.XXX ERROR: lsUtil: srcPath /otherZone/home/bobby#tempZone does not exist or user lacks access permission
```
and in the log file:
```bash
May 22 07:39:01 pid:31660 remote addresses: XXX.XXX.XXX.XXX ERROR: initZone: connection info for zone otherZone not configured
```
Everything works as soon as I add the connection string during the `mkzone` call.

**Comment by korydraughn** [comment]
The **zone** word's casing is inconsistent throughout the doc. How should it appear in the doc? I feel it should be lowercase when used in paragraphs and uppercase when used in titles.

**Comment by korydraughn** [comment]
Also, what do we want to do with this?  Do we want to delete it or move it somewhere else?
```
To configure this, add items to the `/etc/irods/server_config.json` file.

`zone_key` is for the servers [Within A Zone](#within-a-zone), for example:

~~~
"zone_key": "asdf1234",
~~~
```

File: docs/system_overview/federation.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (trel) [reply docs/system_overview/federation.md:74]: dang.  i thought we would be able to remove that... okay - we'll definitely investigate this for 4.2.9 and get it better/consistent.    yes, leave the connection information for now.

**Comment by trel** [comment]
> The **zone** word's casing is inconsistent throughout the doc. How should it appear in the doc? I feel it should be lowercase when used in paragraphs and uppercase when used in titles.

Throughout the documentation I have been capitalizing all our proper nouns (Zone, Resource, etc.)... but in the case of examples... it could be confusing that way.   Let's go with lowercase for now on this page when talking about them, and uppercase when using the example values.

**Comment by trel** [comment]
> Also, what do we want to do with this? Do we want to delete it or move it somewhere else?
> 
> ```
> To configure this, add items to the `/etc/irods/server_config.json` file.
> 
> `zone_key` is for the servers [Within A Zone](#within-a-zone), for example:
> 
> ~~~
> "zone_key": "asdf1234",
> ~~~
> ```

let's remove this section - it doesn't line up anymore - and your example above is more comprehensive and makes the same point.

File: docs/system_overview/federation.md [hunk 5]
```diff
@@ -7,27 +7,81 @@ Primary reasons for using Zone Federation include:
...
+        "zone_key": "ZONE_KEY_FOR_ALL_OF_ZONE_A", 
+        "zone_name": "ZoneA", 
+        "zone_port": 1247
+    }
+]
+~~~
+
+For more information, see [Server Authentication](#server-authentication).
+
+It is important that the servers are able to resolve the host name of the federated zone. This can be handled in many ways. The resolution could come from this account's `hosts_config.json` file, this server's `/etc/hosts` file, or DNS in priority order respectively.
```
Comment (trel) [inline docs/system_overview/federation.md:67]: please update to "host name(s) in the federated zone"

Comment (trel) [reply docs/system_overview/federation.md:67]: please add a comma... "in priority order, respectively"

File: docs/system_overview/federation.md [hunk 6]
```diff
@@ -7,27 +7,81 @@ Primary reasons for using Zone Federation include:
...
+
+With the server configuration and networking complete, all that is left is to define the remote zone and users in the catalog.
```
Comment (trel) [inline docs/system_overview/federation.md:69]: awkward construction... "all that is left is to define"...

how about...

"the final step is to define"

File: docs/system_overview/federation.md [hunk 7]
```diff
@@ -104,7 +159,7 @@ Not having this required additional stanza will likely result in a `REMOTE_SERVE
 
 ## Federation with iRODS 3.x
 
-iRODS 4.0+ has made some additions to the database tables for the resources (`r_resc_main`) and Data Objects (`r_data_main`) for the purposes of tracking resource hierarchy, children, parents, and other relationships.  These changes would have caused a cross-zone query to fail when the target zone is iRODS 3.x.
+iRODS 4.0+ has made some additions to the database tables for the resources (`r_resc_main`) and data objects (`r_data_main`) for the purposes of tracking resource hierarchy, children, parents, and other relationships.  These changes would have caused a cross-zone query to fail when the target zone is iRODS 3.x.
 
 In order to support commands such as `ils` and `ilsresc` across a 3.x to 4.0+ federation, iRODS 4.0+ will detect the cross zone query and subsequently strip out any requests for columns which do not exist in the iRODS 3.x table structure in order to allow the query to succeed.
```
Comment (trel) [inline docs/system_overview/federation.md:164]: "will detect" -> "detects"

**Comment by korydraughn** [comment]
The following statement from **Between Two Zones** makes me think the `zone_key` can be set to anything. It will likely confuse new users and should be changed or removed.
```
The `zone_key` should be a unique and arbitrary string, one for each zone.
```

**Comment by korydraughn** [comment]
> The following statement from **Between Two Zones** makes me think the `zone_key` can be set to anything. It will likely confuse new users and should be changed or removed.
> 
> ```
> The `zone_key` should be a unique and arbitrary string, one for each zone.
> ```

how about ...
```
The `zone_key` must match the remote zone's `zone_key` (the value specified during installation and setup).
```

**Comment by trel** [comment]
> > The following statement from **Between Two Zones** makes me think the `zone_key` can be set to anything. It will likely confuse new users and should be changed or removed.
> > ```
> > The `zone_key` should be a unique and arbitrary string, one for each zone.
> > ```
> 
> how about ...
> 
> ```
> The `zone_key` must match the remote zone's `zone_key` (the value specified during installation and setup).
> ```

that seems good

**Comment by korydraughn** [comment]
Changes pushed.

With that change, please see the last sentence of the first paragraph under **Within a Zone**. This statement makes me think the zone key is optional and possibly not set during installation.

The zone key is required for the server to function properly, right? Should this statement be tweaked as well?

**Comment by trel** [comment]
definitely not optional.   perhaps just adding 'is required and'?...

"This `zone_key` is required and should be a unique and..."

**Comment by korydraughn** [comment]
should i mention that this property is normally set during installation and setup?

**Comment by trel** [comment]
sure thing.

**Comment by trel** [comment]
ready for squashing when you are

**Comment by korydraughn** [comment]
This is what i changed it too:

When a client connects to a consumer server and then authenticates, the consumer server connects to the provider server to perform the authentication. To make this more secure, you must configure the zone_key to cause the iRODS system to authenticate the servers themselves. **This is normally done during installation and setup**. This zone_key is required and should be a unique and arbitrary string (maximum alphanumeric length of 49), one for your whole zone:

**Comment by trel** [comment]
yep - good.   awaiting maximum squash event.

**Comment by korydraughn** [comment]
Added #

**Pull request merged**